clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -146,6 +146,184 @@ class image_mem_wrapper {
   image_mem_wrapper *_sub_wrappers{nullptr};
 };
 
+/// @brief  The wrapper structure for external memory handle desc
+struct external_mem_handle_desc {
```

Comment:
If we don't want the user to access member variables, please use `class` instead of `struct`. Same for other structs.